aptos-move/framework/aptos-framework/sources/multisig_account.move

```diff
@@ -0,0 +1,1348 @@
+        let multisig_account_resource = borrow_global<MultisigAccount>(multisig_account);
+        let transaction_id = multisig_account_resource.last_transaction_id + 1;
+        let transaction = table::borrow(&multisig_account_resource.transactions, transaction_id);
+
+        if (option::is_some(&transaction.payload)) {
+            *option::borrow(&transaction.payload)
+        } else {
+            provided_payload
+        }
+    }
+
+    #[view]
+    public fun can_be_executed(
+        multisig_account: address, transaction_id: u64): bool acquires MultisigAccount {
+        let multisig_account_resource = borrow_global_mut<MultisigAccount>(multisig_account);
+        let transaction = table::borrow(&mut multisig_account_resource.transactions, transaction_id);
+        transaction_id == multisig_account_resource.last_transaction_id + 1 &&
+            transaction.status.status == TRANSACTION_STATUS_PENDING &&
+            simple_map::length(&transaction.approvals) >= multisig_account_resource.signatures_required
+    }
+
+    #[view]
+    public fun can_be_removed(
+        multisig_account: address, transaction_id: u64): bool acquires MultisigAccount {
+        let multisig_account_resource = borrow_global_mut<MultisigAccount>(multisig_account);
+        let transaction = table::borrow(&mut multisig_account_resource.transactions, transaction_id);
+        transaction_id == multisig_account_resource.last_transaction_id + 1 &&
+            transaction.status.status == TRANSACTION_STATUS_PENDING &&
+            simple_map::length(&transaction.rejections) >= multisig_account_resource.signatures_required
+    }
+
+    #[view]
+    public fun get_next_multisig_account_address(creator: address): address {
+        let owner_nonce = account::get_sequence_number(creator);
+        create_resource_address(&creator, create_multisig_account_seed(to_bytes(&owner_nonce)))
+    }
+
+    #[view]
+    /// Return the id of the last transaction that was executed (successful or failed) or removed.
+    public fun last_resolved_transaction_id(multisig_account: address): u64 acquires MultisigAccount {
+        let multisig_account_resource = borrow_global_mut<MultisigAccount>(multisig_account);
+        multisig_account_resource.last_transaction_id
+    }
+
+    /// Creates a new multisig account on top of an existing account.
+    ///
+    /// This offers a migration path for an existing account with a multi-ed25519 auth key (native multisig account).
+    /// In order to ensure a malicious module cannot obtain backdoor control over an existing account, a signed message
+    /// with a valid signature from the account's auth key is required.
+    public entry fun create_with_existing_account(
+        multisig_account: &signer,
+        owners: vector<address>,
+        signatures_required: u64,
+        account_scheme: u8,
+        account_public_key: vector<u8>,
+        create_multisig_account_signed_message: vector<u8>,
```

Comment:
In this signature scenario, it will require two round of signature collections from multiple owners: 
1. Collect the signature for `create_multisig_account_signed_message`
2. Collect the signature for message payload as `&signer`. 

This is not user friendly. Can it be simplified?

Reply:
Agree with the friction there. Any suggestions for improving this? We can remove the signed message here but that would be risky as it can allow an account to be completely taken over by creating a multisig account on top with just the signer.

Reply:
100% Agree. I would suggest to have all the message included in the challenge, including account address, sn, chain id, e.t.c. Can the domain specific signature feature help with this one?

Reply:
Updated the message to include account address (for cross-account replay if same auth key) and sequence_number (more for timely execution guarantee, not for replay protection as multisig account creation can only be done once).